### PR TITLE
Add "allow download" option for VideoPress video

### PR DESCRIPTION
**Pull request by thedebian**
Fixes Automattic/greenhouse#1062

#### Changes proposed in this Pull Request:

This PR adds the "allow download" option for VideoPress videos, allowing the video to be displayed with a "Download video" button linking to the original video.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* On a sandboxed jetpack with jurassic tube and VideoPress enabled
* Sandbox WPCOM public-api.wordpress.com
* Apply the wpcom patch stack
    * phab co D72213
    * arc patch D72220  
* Go to your media library /wp-admin/upload.php
* Select or upload and select a VideoPress video to open the attachment details
* ✅ The following checkbox should be visible

Capture d’écran 2021-12-28 à 15 04 07

* Toggle the checkbox and reload the page
* ✅ The checkbox should be in the same state when reloading the page 
* ✅ No other setting should be affected by this change
* Create or edit an existing post using the block editor
* Add a /video block
* Select the VideoPress video from your media library
* ✅ The "Allow download" toggle should be visible
 
Capture d’écran 2021-12-28 à 15 10 09

* ✅ The "Allow download" toggle should be in the same state as in the media  library
* Toggle the "allow download"  setting, save and reload the page
* ✅ The "allow download" toggle should be in the same state

### CODE REVIEW

File: projects/plugins/jetpack/modules/videopress/class-videopress-attachment-metadata.php [hunk 1]
```diff
@@ -15,27 +15,30 @@ class Videopress_Attachment_Metadata {
...
 
 		$args = array(
 			'method'  => 'POST',
 			'headers' => array( 'content-type' => 'application/json' ),
 		);
 
-		$display_embed = (int) $display_embed;
+		// Keep null values to avoid accidental unset 
+		$display_embed  = null === $display_embed ? null : (int) $display_embed;
+		$allow_download = null === $allow_download ? null : (int) $allow_download;
```
Comment (thedebian) [inline projects/plugins/jetpack/modules/videopress/class-videopress-attachment-metadata.php:39]: This fixes an existing bug on Calypso, where checking one of the box would unset the other (or only updating the rating too).

**Review by pgk — CHANGES_REQUESTED** [review]
Tested well for me on both the media library and the editor! A few wee changes we can do for clarity.

**Review by pgk — APPROVED** [review]
lgtm!

**Review by jeherve — APPROVED** [review]
This tests well for me. 🚢

**Comment by thedebian** [comment]
Changeset 237513

**Pull request merged**